src/crypto/SymmetricCipher.cpp

```diff
@@ -176,7 +176,7 @@ SymmetricCipher::Mode SymmetricCipher::stringToMode(const QString& cipher)
         return Aes128_CTR;
     } else if (cipher.compare("aes-256-ctr", cs) == 0 || cipher.compare("aes256-ctr", cs) == 0) {
         return Aes256_CTR;
-    } else if (cipher.compare("aes-256-gcm", cs) == 0 || cipher.compare("aes256-gcm", cs) == 0) {
+    } else if (cipher.compare("aes-256-gcm", cs) == 0 || cipher.startsWith("aes256-gcm", cs)) {
```

Comment:
This is a very specific change for openssh, I would rather we remove the suffix that openssh places on the cipher string before passing it into this function, line 313 of OpenSSHKey.cpp.